lib/async_hooks.js

```diff
@@ -206,34 +206,71 @@ class AsyncResource {
   }
 }
 
-const locals = [];
+let activeAsyncLocalStorageCount = 0;
+const kLocalStorageList = Symbol('kLocalStorageList');
 const asyncContextHook = createHook({
   init(asyncId, type, triggerAsyncId, resource) {
-    for (let i = 0; i < locals.length; ++i) {
-      locals[i]._propagate(resource, executionAsyncResource());
+    const triggerResource = executionAsyncResource();
```

Comment:
nit: `triggerResource` is confusing here as it doesn't match the `triggerId`. Maybe use `executionResource` or `currentResource`.

Reply:
Done in https://github.com/nodejs/node/pull/26540/commits/8bbe84d3859cb3b7dc899b5cc32a12f4220e2cb1